### DISCUSSION TITLE
Add support for Ascend NPU

### DISCUSSION THREAD

**Discussion by qnxgy921** [openai/whisper#2223] [Ideas]
In China, Ascend NPU is the second choice after Nvidia GPU and has been adpoted by many companies, such as Alibaba, ByteDance, Meituan, etc. Huawei officially released an adapter called [torch_npu](https://github.com/Ascend/pytorch/blob/master/README.en.md), to adapt PyTorch on Ascend NPU. torch_npu is user friendly to developers, so that we can still enjoy the same PyTorch experience that we accustomed to today.

Since the NPU is supported in Transformers and Accelerate, Is there any way to deploy whisper on NPU?
Reference: [https://github.com/huggingface/transformers/issues/22600](url)

**Comment by phineas-pta** [comment]
u have to write your own python code to load whisper in pytorch npu
